google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb

```diff
@@ -360,6 +360,21 @@ def self.derive_source_format path
           return "DATASTORE_BACKUP"       if path.end_with? ".backup_info"
           nil
         end
+
+        ##
+        # @private
+        #
+        # Converts a primitive time value in milliseconds to a Ruby Time object.
+        #
+        # @return [Time, nil] The Ruby Time object, or nil if the given argument
+        #   can't be converted to an Integer.
+        def self.millis_to_time time_millis
+          time_millis = Integer time_millis
+          time_secs = time_millis / 1000.0
+          ::Time.at time_secs
+        rescue StandardError
```

Comment:
I wonder if it wouldn't be better to check for a `nil` value before calling `Integer time_millis`, which will raise if `time_millis` isn't an integer value, instead of rescuing here.

Reply:
What was the reason for the original `rescue`? Just a rubocop rule, or is there a possibility of some value other than `nil` that can't be converted?

Reply:
I honestly don't remember.

Reply:
To me it looks like control flow. And we shouldn't be using exceptions for control flow, where it can be helped.